interfaces/cython/cantera/examples/reactors/ic_engine.py

```diff
@@ -54,200 +56,203 @@
-        test[n1] = 1
-    else:
-        inlet_valve.valve_coeff = 0.
-    if (np.mod(crank_angle(t_i) - outlet_open, 4 * np.pi) <
-            np.mod(outlet_close - outlet_open, 4 * np.pi)):
-        outlet_valve.valve_coeff = outlet_valve_coeff
-    else:
-        outlet_valve.valve_coeff = 0.
-    if (np.mod(crank_angle(t_i) - injector_open, 4 * np.pi) <
-            np.mod(injector_close - injector_open, 4 * np.pi)):
-        injector_mfc.set_mass_flow_rate(injector_mass / injector_t_open)
-    else:
-        injector_mfc.set_mass_flow_rate(0)
-
-    # perform time integration, refine time step if necessary
-    solved = False
-    for n2 in range(4):
-        try:
-            sim.advance(t_i)
-            solved = True
-            break
-        except ct.CanteraError:
-            sim.set_max_time_step(1e-6 * 10. ** -n2)
-            n_last_refinement = n1
-    if not solved:
-        raise ct.CanteraError('Refinement limit reached')
-    # coarsen time step if too long ago
-    if n1 - n_last_refinement == n_wait_coarsening:
-        sim.set_max_time_step(1e-5)
-
-    # write output data
-    states.append(r.thermo.state)
-    V[n1] = r.volume
-    m[n1] = r.mass
-    mdot_in[n1] = inlet_valve.mdot(0)
-    mdot_out[n1] = outlet_valve.mdot(0)
-    d_W_v_d_t[n1] = - (r.thermo.P - ambient_air.thermo.P) * A_piston * \
-        piston_speed(t_i)
-    heat_release_rate[n1] = - r.volume * ct.gas_constant * r.T * \
-        np.sum(gas.standard_enthalpies_RT * r.thermo.net_production_rates, 0)
-
-
-#####################################################################
+states = ct.SolutionArray(cyl.thermo,
+                          extra=('t', 'ca', 'V', 'm',
+                                 'mdot_in', 'mdot_out',
+                                 'dWv_dt', 'heat_release_rate'))
+
+# simulate with a minimum resolution of 1 deg crank angle
+dt = 1. / (360 * f)
+t_stop = sim_n_revolutions / f
+while sim.time < t_stop:
+
+    # perform time integration
+    sim.advance(sim.time + dt)
+
+    # calculate results to be stored
+    dWv_dt = - (cyl.thermo.P - ambient_air.thermo.P) * A_piston * \
+        piston_speed(sim.time)
+    heat_release_rate = - cyl.volume * ct.gas_constant * cyl.T * \
+        np.sum(gas.standard_enthalpies_RT * cyl.thermo.net_production_rates, 0)
+
+    # append output data
+    states.append(cyl.thermo.state,
+                  t=sim.time*1e3, ca=crank_angle(sim.time),
+                  V=cyl.volume, m=cyl.mass,
+                  mdot_in=inlet_valve.mdot(sim.time),
+                  mdot_out=outlet_valve.mdot(sim.time),
+                  dWv_dt=dWv_dt,
+                  heat_release_rate=heat_release_rate)
+
+t = 1.e-3 * states.t
+
+# #####################################################################
 # Plot Results in matplotlib
-#####################################################################
+# #####################################################################
```

Comment:
The spaces like `# #` seem weird to me. I presume this is an autopep8 thing, but since these are intended to be separators, I don't think the space makes sense.